mitmproxy/tools/web/app.py

```diff
@@ -324,6 +327,59 @@ def broadcast(cls, **kwargs):
 class ClientConnection(WebSocketEventBroadcaster):
     connections: ClassVar[set] = set()
 
+    def __init__(self, application: Application, request, **kwargs):
+        super().__init__(application, request, **kwargs)
+        self.application = application
+        self.filters: dict[str, str] = {}  # Instance-level filters
```

Comment:
```suggestion
        self.filters: dict[str, TFilter] = {}  # Instance-level filters
```

(this way we don't have to re-parse all the time, and we indirectly ensure that we only get valid filter exprs in there)